### ISSUE TITLE
Incompatible with haya14busa/incsearch.vim

### ISSUE THREAD

**Issue by davidcgl** [google/vim-searchindex#6]
Custom star commands (*, #) doesn't work well with [haya14busa/incsearch.vim](https://github.com/haya14busa/incsearch.vim) with the following mappings:
map /  <Plug>(incsearch-forward)
map ?  <Plug>(incsearch-backward)
map g/ <Plug>(incsearch-stay)

I have both plugins installed, and repeatedly pressing \* or # selects the next / previous word instead of word under cursor, as shown below.

![recording](https://cloud.githubusercontent.com/assets/1164687/10591939/cc28d772-766d-11e5-8d9d-588378f9c13d.gif)

**Comment by rburny** [comment]
TL;DR: `:let g:searchindex_improved_star=0`

The explanation is as follows: because Vim's implementation of "star" commands is hacky (it silently ignores some options), vim-searchindex reimplements them; see `:searchindex-star`. By remapping slash, you broke the reimplementation. Disabling reimplementation will make the issues goes away, there are only two minor down-sides:
- occasional miscounting (only if you use `smartcase` option, and only on "star" commands)
- you won't be able to use `g:searchindex_star_case` option

It might be possible to work around the problem, but I'm not sure it's worthwhile, given that these two plugins conflict anyway. For example, a much bigger problem is that you remap normal search (/ and ?) and they no longer show search index. This is because incsearch uses fake command line, and searchindex uses "real" one. Fixing this seems to require adding hooks to incsearch.

**Comment by rburny** [comment]
After all, I've found that it was a problem with `searchindex` using `map` instead of `noremap`. You should now be able to use the two plugins together, even with  `g:searchindex_improved_star=1`.

**Comment by Frederick888** [comment]
It seems that searchindex also needs to load *after* vim-asterisk, otherwise searchindex's counter shows wrong numbers all the time.

```vim
" part of my vimrc. I didn't map /, ?, etc.

call plug#begin('~/.vim/plugged')
Plug 'haya14busa/vim-asterisk'
Plug 'google/vim-searchindex'
call plug#end()

" vim-asterisk
let g:asterisk#keeppos = 1
map *  <Plug>(asterisk-z*)
map #  <Plug>(asterisk-z#)
map g* <Plug>(asterisk-gz*)
map g# <Plug>(asterisk-gz#)
map z*  <Plug>(asterisk-*)
map z#  <Plug>(asterisk-#)
map gz* <Plug>(asterisk-g*)
map gz# <Plug>(asterisk-g#)
```

Edit: I somehow had to swap them recently...
